res_handler_jq_transform.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/mitchellh/mapstructure"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+type ResponseTransformJQOptions struct {
+	//FlushInterval time.Duration
+}
+
+type ResponseTransformJQMiddleware struct {
+	Spec   *APISpec
+	config ResponseTransformJQOptions
+}
+
+func (h *ResponseTransformJQMiddleware) Init(c interface{}, spec *APISpec) error {
+	h.Spec = spec
+	if err := mapstructure.Decode(c, &h.config); err != nil {
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (h *ResponseTransformJQMiddleware) HandleResponse(rw http.ResponseWriter, res *http.Response, req *http.Request, ses *SessionState) error {
+	_, versionPaths, _, _ := h.Spec.GetVersionData(req)
+	found, meta := h.Spec.CheckSpecMatchesStatus(req, versionPaths, TransformedJQResponse)
+	if !found {
+		return nil
+	}
+
+	// Read the body:
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	t := meta.(*TransformJQSpec)
+
+	var bodyObj map[string]interface{}
+	if err := json.Unmarshal(body, &bodyObj); err != nil {
```

Comment:
As before, use `r.Body` directly.

Reply:
Actually, the other two comments apply here as well, as there is some degree of copy-pasting.